### PR TITLE
[PO-31] SignedBlockHeader

### PR DESCRIPTION
**Pull request by mirkoAlic**
The goal of this PR is allow the client signs block headers, and sends them to others.

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/network/p2p/messages/PV62.scala [hunk 1]
```diff
@@ -108,55 +109,91 @@ object PV62 {
...
+          )
+      }
+    }
+  }
+
+  object SignedBlockHeaders {
+
+    val code: Int = Versions.SubProtocolOffset + 0x04
+
+    implicit class BlockHeadersEnc(val underlyingMsg: SignedBlockHeaders) extends
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/network/p2p/messages/PV62.scala]: Very minor detail: Maybe we should rename this and `BlockHeadersDec` as `SignedBlockHeadersEnc`?

File: src/main/scala/io/iohk/ethereum/ledger/Ledger.scala [hunk 2]
```diff
@@ -585,19 +588,19 @@ class LedgerImpl(
...
 
-    val minerAddress = Address(block.header.beneficiary)
+    val minerAddress = Address(block.signedHeader.header.beneficiary)
     val minerAccount = getAccountToPay(minerAddress, worldStateProxy)
-    val minerReward = blockRewardCalculator.calcBlockMinerReward(block.header.number, block.body.uncleNodesList.size)
-    val afterMinerReward = worldStateProxy.saveAccount(minerAddress, minerAccount.increaseBalance(UInt256(minerReward)))
-    log.debug(s"Paying block ${block.header.number} reward of $minerReward to miner with account address $minerAddress")
-
-    block.body.uncleNodesList.foldLeft(afterMinerReward) { (ws, ommer) =>
+    val minerReward = blockRewardCalculator.calcBlockMinerReward(block.signedHeader.header.number, block.body.uncleNodesList.size)
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/ledger/Ledger.scala]: As we have removed the whole usage of ommers, maybe we should temporarely pass 0 here as the ommer length?

Comment (mirkoAlic) [reply src/main/scala/io/iohk/ethereum/ledger/Ledger.scala]: sure!

File: src/main/scala/io/iohk/ethereum/network/p2p/messages/PV62.scala (hunk 1 above)
Comment (mirkoAlic) [reply src/main/scala/io/iohk/ethereum/network/p2p/messages/PV62.scala]: :+1:

**Review by ntallar — APPROVED** [review]
LGTM!

**Pull request merged**